cam.go

```diff
@@ -393,16 +451,17 @@ func (fc *filteredCamera) shouldSend(ctx context.Context, img image.Image) (bool
 			match, label := fc.anyDetectionsMatch(vs.Name().Name, res, false)
 			if match {
 				fc.logger.Debugf("keeping image with objects %v", res)
-				fc.buf.MarkShouldSend(fc.conf.WindowSeconds)
+				fc.buf.MarkShouldSend(now)
 				fc.acceptedStats.update(label.Label())
 				return true, nil
 			}
 		}
-
-		if time.Now().Before(fc.buf.CaptureTill) {
-			// send, but don't update captureTill
-			return true, nil
-		}
+	}
+	// if we don't inhibit or update CaptureTill with another trigger, check to see if still within
+	// the CaptureTill window
```

Comment:
moved this to outside the loop, so that we go check through every vision service before deciding not update captureTill